pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,205 @@
+        - `ORGANIZATION_SLUG` can be obtained:
+
+            * From the end of your Buildkite URL, after accessing **Pipelines** in the global navigation of your organization in Buildkite.
+
+            * By running the [List organizations](/docs/apis/rest-api/organizations#list-organizations) REST API query to obtain this value from `slug` in the response. For example:
+
+                ```bash
+                curl - X GET "https://api.buildkite.com/v2/organizations" \
+                  -H "Authorization: Bearer $TOKEN"
+                ```
+        - `PIPELINE_SLUG` can be obtained:
+
+            * From the end of your Buildkite URL, after accessing **Pipelines** in the global navigation of your organization in Buildkite, then accessing the specific pipeline to be specified in the custom trust policy.
+
+            * By running the [List pipelines](/docs/apis/rest-api/pipelines#list-pipelines) REST API query to obtain this value from `slug` in the response from the specific pipeline. For example:
+
+                ```bash
+                curl - X GET "https://api.buildkite.com/v2/organizations/{org.slug}/pipelines" \
+                  -H "Authorization: Bearer $TOKEN"
+                ```
+        - `REF` is usually replaced with `refs/heads/main` to enforce the IAM role's access and use to only the `main` branch, `refs/tags/*` to ensure only tagged releases are able to be deployed, or a wildcard `*` if the IAM role can be accessed and used by all branches.
+        - `BUILD_COMMIT` (optional) can be omitted and if so, is usually replaced with a single wildcard `*` at the end of the line.
+        - `STEP_KEY` (optional) can be omitted and if so, is usually replaced with a single wildcard `*` at the end of the line.
+
+    **Note:** When formulating your _subject_ field's value, you can replace any of the constituent field values above with a wildcard `*` to not set limits on those constituent fields.
+
+    For example, to allow the IAM role to be used for any pipeline in the Buildkite organization `example-org`, when building on any branch or tag, specify a single _subject_ field value of `organization:example-org:*`.
+
+    You can also allow this IAM role to be used with other pipelines, branches, commits and steps by specifying multiple comma-separated values for the `agent.buildkite.com:sub` _subject_ field.
+
+1. Modify the `Condition` section's `IpAddress` values (`AGENT_PUBLIC_IP_ONE` and `AGENT_PUBLIC_IP_TWO`) with a list of your agent's IP addresses or CIDRs.
+
+    Only OIDC token exchange requests (for IAM roles) from Buildkite Agents with these IP addresses will be permitted.
```

Comment:
It's worth pointing out this only works if you have dedicated public IPs. One of the options our elastic stack gives is to run agents in public subnets with ephemeral IPs.
We could cross link to the elastic-stack documentation to let them know how to do this.

Reply:
Need to incorporate relevant static IP configuration content in the relevant pages within https://buildkite.com/docs/agent/v3/elastic-ci-aws/elastic-ci-stack-overview first before we can mention how to configure dedicated IPs in elastic stack.